src/main/java/com/seller/panel/service/LoginService.java

```diff
@@ -0,0 +1,32 @@
+package com.seller.panel.service;
+
+import com.seller.panel.dto.LoginRequest;
+import com.seller.panel.model.Users;
+import com.seller.panel.util.JwtTokenUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+public class LoginService extends BaseService{
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private JwtTokenUtil jwtTokenUtil;
+
+    public String validateDataAndGenerateToken(LoginRequest loginRequest) {
```

Comment:
A function should not perform two operations, having said that here validation is an operation not data validation of values passed to the service function.
   I don't think you need login service at all, call `authenticate(email, password) (function name more apt)` from controller and send the claims to JWTUtil from there only to generate token.

Reply:
Also, here you would need one extra test case for this service, which you have already tested in `Userservice` and `JWTToken`, now you would write controllers unit test also, so total 4, you can have 1 less  😃